src/kernels/kernelproduct.jl

```diff
@@ -29,23 +29,26 @@ hadamard(x,y) = x.*y
 function kernelmatrix(
     κ::KernelProduct,
     X::AbstractMatrix;
-    obsdim::Int=defaultobs)
-    reduce(hadamard,kernelmatrix(κ.kernels[i],X,obsdim=obsdim) for i in 1:length(κ))
+    obsdim::Int=defaultobs,
+)
+    reduce(hadamard, kernelmatrix(κ.kernels[i], X, obsdim = obsdim) for i in 1:length(κ))
```

Comment:
Typically
```suggestion
    return mapreduce(hadamard, kernels(κ)) do kernel
        kernelmatrix(kernel, X; obsdim = obsdim)
    end
```
should be more efficient since as the documentation says
```julia
mapreduce is functionally equivalent to calling reduce(op, map(f, itr); init=init), but will in general execute faster since no intermediate collection needs to be created.
```

Reply:
But again one maybe should do that in a separate PR :smile: